crates/store/src/state_db.rs

```diff
@@ -45,17 +44,43 @@ impl StateDBVersion {
     pub fn is_genesis_version(&self) -> bool {
         self.block_hash.is_none() && self.tx_index.is_none()
     }
+
+    fn lock_block_number_and_tx_index(&self, db: &StoreTransaction) -> Result<(u64, u32), Error> {
+        if self.is_genesis_version() {
+            return Ok((0u64, 0u32));
+        }
+        let block_hash = &self
+            .block_hash
+            .ok_or_else(|| "Block hash doesn't exist".to_owned())?;
+        let block_number = db
+            .get_block_number(block_hash)?
+            .ok_or_else(|| "Block number doesn't exist".to_owned())?;
+        let block = db
+            .get_block(block_hash)?
+            .ok_or_else(|| "Block doesn't exist".to_owned())?;
+        let tx_index = match self.tx_index {
+            Some(tx_index) => {
+                if block.transactions().len() <= tx_index as usize {
```

Comment:
```suggestion
                if block.transactions().len() >= tx_index as usize {
```

Reply:
Why we still pass the tests?

Reply:
State db's unit test's data are all be injected by the constructor from_tx_index, that means 
decoupling with the data of block, tx and other domain related.

This case (except equal case) is may not covered by all tests.

Reply:
I will try to make it up somewhere.